### PR TITLE
you can no longer cryopod others

### LINKED ISSUE #10993: Players can cryopod players again
I am going to stab a mother fucker.

### PR DESCRIPTION
**Pull request by ynot01**
# General Documentation

### Intent of your Pull Request

cryoing other players is now impossible
admin-cryo will still work because it doesn't use the mousedrop-t proc
also fixes the bug where you can cryo catatonics because jamies code didn't work

### Why is this change good for the game?

prevents players from griefing each other

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
cryoing other players is now impossible

# Changelog

:cl:  
tweak: You can no longer put other players into cryopods
/:cl:

### CODE REVIEW

**Comment by boodaliboo** [comment]
not against this but was this a issue

**Comment by ChesterTheCheesy** [comment]
tfw can't cryo my ssd blood bro with no admin online anymore

**Comment by MegaEmpirical** [comment]
> not against this but was this a issue

I've been cryoed when dealing with tickets while aghosted. This is supposedly fixed though, independent of this.

I also had a ticket a week or two ago where someone had to restart their internet and they came back to a different player being in their body because someone shoved them in.

Also, currently catatonic clone copies can be put in and get a ghost in them.

So it has happened and can happened, the latter of which is the important one. We also get tickets about people saying they need to do X for 30 minutes and will be back, we as admins are aware of this and can handle it correctly, players have no idea why someone is brain dread and don't have that context.

**Comment by Sniblet** [comment]
Wait you mean I wasn't supposed to do that

**Comment by Cartlord** [comment]
Fixes https://github.com/yogstation13/Yogstation/issues/10993 , no?

**Comment by burn1cycle** [comment]
Against. 
I'm for tweaking it so people who get put into cryostorage get a 10 minute grace period to come back and AGhosts(admin ghosts) get a special "Do not store" status(this was always how it had been intended to function, I'm not sure what happened.), but(except for AGhosts) if they are going to be absent from the round for more than 10 minutes then it's problematic for every other player who is actively participating in the round.

-  Inactivity has a strong effect on our game.
-  Inactive players are taking up a role slot in some cases, 
- They get looted and since the player never returns to the game the admin is never asked to investigate and it's rare that I've seen an admin proactively open up a ticket on their own unless it was large scale grief. 
- Various antags can gain benefits from SSD persons(again going unreported)
- SSD persons don't stop being a target until cryo'd and lacking an active person to antag ruins an antag's round.
- Inactive players give an incorrect impression of what the crew can deal with
- Inactive players catch diseases that often go uncured.
 
  Basically, when you click "join" and enter the game we should expect you to commit to that role for the shift or else leave in an appropriate way, failing that we should be able to put you in cryo.

  **I lack faith in a coder decision effecting administrative policy**. I doubt we will have strict and uniform enforcement of what admins will do with SSD people or if some admins will even care at all and that's only if they are online at the time. 




This is off topic but  I don't particularly like the functionality that allows ghosts to assume cryostorage roles since they may have loads of OOC knowledge from being a ghost that's hard or impossible to ignore.

File: code/game/machinery/cryopod.dm [hunk 1]
```diff
@@ -364,11 +364,8 @@ GLOBAL_LIST_INIT(typecache_cryoitems, typecacheof(list(
 		to_chat(user, "<span class='notice'>This person cannot be put in cryogenic storage!</span>")
 		return
 
-	if(target.client && user != target)
-		if(iscyborg(target))
-			to_chat(user, "<span class='danger'>You can't put [target] into [src]. They're online.</span>")
-		else
-			to_chat(user, "<span class='danger'>You can't put [target] into [src]. They're conscious.</span>")
+	if(user != target)
+		to_chat(user, "<span class='danger'>You are unable to cryo others.</span>")
```
Comment (alexkar598) [inline code/game/machinery/cryopod.dm:368]: Cryo is not a verb

File: code/game/machinery/cryopod.dm [hunk 2]
```diff
@@ -399,11 +396,10 @@ GLOBAL_LIST_INIT(typecache_cryoitems, typecacheof(list(
 
 	if(target == user)
```
Comment (alexkar598) [inline code/game/machinery/cryopod.dm:397]: Check will always return true

**Review by alexkar598 — COMMENTED** [review]
Verify that target is equal to user and remove references to user. Also clear out checks for clientless in close_machine() as the user will always have a client

**Review by alexkar598 — CHANGES_REQUESTED** [review]
meant to be request changes

**Comment by burn1cycle** [comment]
Even more against after some experience trying to tell admins players are SSD and need to be taken care of. Got the response "Dude? What's going on? Is this an admin issue?". Fix the admins first then fix cryopods.

**Comment by MegaEmpirical** [comment]
> Even more against after some experience trying to tell admins players are SSD and need to be taken care of. Got the response "Dude? What's going on? Is this an admin issue?". Fix the admins first then fix cryopods.

The issue is the admin there. Who was it?

**Comment by burn1cycle** [comment]
> > Even more against after some experience trying to tell admins players are SSD and need to be taken care of. Got the response "Dude? What's going on? Is this an admin issue?". Fix the admins first then fix cryopods.
> 
> The issue is the admin there. Who was it?

I can't remember and it doesn't matter. The only time you're caring to enforce it is when a PR is at stake. I'm not going to eat up my round time and tick off admins in forum complaints who will inevitably drop the excuse:

> limits to complexity that we can be effective on at once. The chaos is worse behind the scenes, and sometimes they cards land where they fall. Some things, simply, cannot be sorted in the exact moment they're happening. If you'd like to try yourself, staff applications are open.

and not be held responsible for anything they failed to do.

 What problem is this solving? That someone was disconnected for a length of time that we've deemed it necessary that they participate in a round? Essentially the PR is just a dodge around participation rules.

**Comment by adamsong** [comment]
There are requested changes that have not been address in a while, and a pull request been merged since that attempts to address this issue in a different way. Some people still want this merged, so I won't close it now, but it is stale.

**Comment by Moja10** [comment]
All cryo needs is a long period of time where they can personally sit in the cryo pod, and if they don't come back in an alloted time then they are removed from the round. If you don't come back in 30 minutes you are out of the round, and if you need more time lock yourself in dorms. Solutions like this PR aren't the answer.

**Comment by MegaEmpirical** [comment]
30 minutes would be way too high. 15 should be okay.

**Comment by alexkar598** [comment]
https://github.com/yogstation13/Yogstation/blob/master/code/game/machinery/cryopod.dm#L173-L174

**Comment by alexkar598** [comment]
That sounds like a great idea! Which is why its already been implemented.

**Comment by alexkar598** [comment]
Stale
